src/sparsezoo/utils/authentication.py

```diff
@@ -124,24 +55,63 @@ def get_auth_header(
 
     Currently only 'public' authentication type is supported.
 
-    :param authentication_type: authentication type for generating token
     :param force_token_refresh: forces a new token to be generated
     :return: An authentication header with key 'nm-token-header' containing the header
         token
     """
-    credentials = SparseZooCredentials()
-    token = credentials.token
-    if token and not force_token_refresh:
-        return {NM_TOKEN_HEADER: token}
-    elif authentication_type.lower() == PUBLIC_AUTH_TYPE:
+    token = _maybe_load_token(path)
+    if token is None or force_token_refresh:
         _LOGGER.info("Obtaining new sparse zoo credentials token")
-        created = time.time()
         response = requests.post(
             url=AUTH_API, data=json.dumps({"authentication_type": PUBLIC_AUTH_TYPE})
         )
         response.raise_for_status()
         token = response.json()["token"]
-        credentials.save_token(token, created)
-        return {NM_TOKEN_HEADER: token}
-    else:
-        raise Exception(f"Authentication type {PUBLIC_AUTH_TYPE} not supported.")
+        created = time.time()
+        _save_token(token, created, path)
+    return {NM_TOKEN_HEADER: token}
+
+
+def _maybe_load_token(path: str):
+    if not os.path.exists(path):
+        _LOGGER.debug(f"No sparse zoo credentials files found at {path}")
+        return None
+
+    _LOGGER.debug(f"Loading sparse zoo credentials from {path}")
+
+    with open(path) as fp:
+        creds = yaml.safe_load(fp)
+
+    if creds is None or CREDENTIALS_YAML_TOKEN_KEY not in creds:
+        _LOGGER.debug(f"No sparse zoo credentials found at {path}")
+        return None
+
+    info = creds[CREDENTIALS_YAML_TOKEN_KEY]
+    if "token" not in info or "created" not in info:
+        _LOGGER.debug(f"No sparse zoo credentials found at {path}")
+        return None
+
+    date_created = datetime.fromtimestamp(info["created"], tz=timezone.utc)
+    creation_difference = datetime.now(tz=timezone.utc) - date_created
+
+    if creation_difference.days > 30:
```

Comment:
docstring for `get_auth_header()` cites that tokens older than 1 day will be updated. Code or doc should be updated